types/react/test/tsx.tsx

```diff
@@ -150,12 +150,10 @@ class ComponentWithLargeState extends React.Component<{}, Record<'a'|'b'|'c', st
     }
 }
 
-class ComponentWithBadLifecycle extends React.Component<{}, {}, number> {
-    getSnapshotBeforeUpdate() { // $ExpectError
-        return 'number';
-    }
-
-    componentDidUpdate(prevProps: {}, prevState: {}, snapshot?: string) { // $ExpectError
-        return;
-    }
-}
+const componentWithBadLifecycle = new (class extends React.Component<{}, {}, number> {})({});
+componentWithBadLifecycle.getSnapshotBeforeUpdate = () => { // $ExpectError
+    return 'number';
+};
+componentWithBadLifecycle.componentDidUpdate = (prevProps: {}, prevState: {}, snapshot?: string) => { // $ExpectError
+    return;
+};
```

Comment:
This is weird but ensures that both TS 2.6 and 2.7 generate the error message in the same line. 2.6 generates it on `class extends`, 2.8 generates it on the individual methods; I assume 2.7 does as well.